src/test/mod.rs

```diff
@@ -428,6 +428,58 @@ fn stdin_works_with_modified_lines() {
     assert_eq!(buf, output.as_bytes());
 }
 
+/// Ensures that `EmitMode::Json` works with input from `stdin`.
+#[test]
+fn stdin_works_with_json() {
+    init_log();
+    let input = "\nfn\n some( )\n{\n}\nfn main () {}\n";
+    #[rustfmt::skip]
+    let output = r#"[{"name":"stdin","mismatches":[{"original_begin_line":1,"original_end_line":6,"expected_begin_line":1,"expected_end_line":2,"original":"\nfn\n some( )\n{\n}\nfn main () {}","expected":"fn some() {}\nfn main() {}"}]}]"#;
```

Comment:
Re: your question on the other thread, ome of the tests for these emitters are already leveraging a separate file with the json/xml/etc. output for the exp values.

For example: https://github.com/rust-lang/rustfmt/blob/954383878e520c3da1cbca4de8eee9d900b986db/src/test/mod.rs#L216-L222 

Those files with the exp output reside in https://github.com/rust-lang/rustfmt/tree/master/tests/writemode/target

Reply:
Thanks - I've added an `assert_stdin_output` which is similar and moved those ugly string literals into test files.